torch/_dynamo/guards.py

```diff
@@ -518,7 +518,7 @@ def __init__(self, expr_to_tensor_ref, id_to_name_map):
         self.id_to_name_map = id_to_name_map
 
     def _print_Symbol(self, expr) -> str:
-        assert isinstance(expr, sympy.core.symbol.Symbol)
```

Comment:
Are you sure this is correct?

Reply:
I think this is due to changes in shape env, can you check that its in either? just to be safe for now.

Reply:
Yeah, I think it's correct because Symbol is a first-class citizen in sympy. It's unlikely sympy will change that. "sympy.Symbol" is safe to use in that sense. 

The previous version "sympy.core.symbol.Symbol" can pass OSS CI but fails when testing on internal dev server. The root cause is that in Sympy 1.4 (the version on dev server) "sympy.core" actually refers to "sympy.core.core", which causes attribute access error for core. Changing to sympy.Symbol fixes it.

I did some testing both internally and on aws cluster. For internal testing, I've tried this replacement and it can fix the previously failed tests (dynamic_shape_test_modules). For OSS, I run the tests with "python test/dynamo/test_dynamic_shapes.py" after the replacement, the test shows OK and it also passes our OSS CI.

Reply:
This is confirmed safe.